cloud-init has 3 failing tests on unstable, causing builds to fail

**Describe the bug**
cloud-init has 3 failing tests on unstable, causing builds to fail. This is possibly related to the Python 3.9 upgrade.

```
>       assert {'primary': expected} == ret
E       AssertionError: assert {'primary': [...ke-1/ubuntu']} == {'primary': [...ake1/ubuntu']}
E         Differing items:
E         {'primary': ['http://fk-fake-1/ubuntu']} != {'primary': ['http://fkfake1/ubuntu']}
E         Use -v to get the full diff
```
[nix-build--TestRunNixosUnstable.txt](https://github.com/NixOS/nixpkgs/files/6863359/nix-build--TestRunNixosUnstable.txt)

**To Reproduce**
Steps to reproduce the behavior:
1. try using cloud-init on unstable

**Expected behavior**
cloud-init is pulled from hydra's cache

**Additional context**
Add any other context about the problem here.

**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.40, NixOS, 21.11pre301044.87807e64a5e (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.12`
 - channels(root): `"nixos-21.11pre301044.87807e64a5e, home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: cloud-init
# a list of nixos modules affected by the problem
module: cloud-init
```

--- a/pkgs/tools/virtualization/cloud-init/default.nix
+++ b/pkgs/tools/virtualization/cloud-init/default.nix
@@ -76,6 +76,8 @@ buildPythonApplication rec {
     "TestConsumeUserDataHttp"
     # Chef Omnibus
     "TestInstallChefOmnibus"
+    # https://github.com/canonical/cloud-init/pull/893
+    "TestGetPackageMirrorInfo"
   ];
 
   disabledTestPaths = [

